fastestimator/dataset/batch_dataset.py

```diff
@@ -307,11 +307,13 @@ def fe_reset_ds(self, shuffle: bool = True, *, seed: Optional[int] = None) -> No
         if self.probability:
             num_samples = num_samples * len(self.datasets)
         self.index_maps = []
-        for dataset, num_sample in zip(self.datasets, num_samples):
+        for idx, (dataset, num_sample) in enumerate(zip(self.datasets, num_samples)):
```

Comment:
we need to prevent BatchDataset from taking an InterleaveDataset as input. Same for ExtendDataset and any other meta datasets

Reply:
BatchDataset I agree, ExtendDataset should be ok